File: code_examples/4-Functions/6-By_Value_or_By_Reference/JavaScript/main.js
/*
Created by: Mr. Coxall
Created on: Sep 2020
This module uses a function by reference
*/

function addOne(someVariable) {
    // The addOne() function adds 1, by reference
    someVariable[0] = someVariable[0] + 1;
}
  
const prompt = require('prompt-sync')()
// In JavaScript, you can pass a variable by reference by using an array  
const someNumber = [];

// Input
const tempVar = parseInt(prompt("Enter a number: "));
someNumber.push(tempVar);

// Process
addOne(someNumber);

// Output
console.log(`The number plus one is: ` + someNumber[0]);

console.log("\nDone.");

File: code_examples/4-Functions/6-By_Value_or_By_Reference/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a function by reference
 */

package main

import "fmt"

func addOne(someVariable *int) {
	// this function adds one to the variable passed in, by reference
	*someVariable = *someVariable + 1
}

func main() {
	// this function uses a function by reference
	var someNumber int

	// input
	fmt.Print("Enter a number: ")
	fmt.Scan(&someNumber)

	// Process
	addOne(&someNumber)

	// Output
	fmt.Println("The number plus one is:", someNumber)

	fmt.Println("\nDone.")
}
